backend/src/data_entry/mod.rs

```diff
@@ -236,62 +216,96 @@ pub async fn polling_station_data_entry_delete(
     ),
 )]
 pub async fn polling_station_data_entry_finalise(
-    State(pool): State<SqlitePool>,
     State(polling_station_data_entries): State<PollingStationDataEntries>,
+    State(elections_repo): State<Elections>,
     State(polling_stations_repo): State<PollingStations>,
-    State(elections): State<Elections>,
     Path((id, entry_number)): Path<(u32, EntryNumber)>,
 ) -> Result<(), APIError> {
-    let polling_station = polling_stations_repo.get(id).await?;
-    let election = elections.get(polling_station.election_id).await?;
-
-    let mut tx = pool.begin().await?;
-
-    let (_, data, _, _) = polling_station_data_entries
-        .get(&mut tx, id, entry_number)
-        .await?;
-    let results = serde_json::from_slice::<PollingStationResults>(&data)?;
+    let state = polling_station_data_entries.get_or_default(id).await?;
 
-    // check that this data entry does not have errors
-    if validate_polling_station_results(&results, &polling_station, &election)?.has_errors() {
-        return Err(APIError::Conflict(
-            "Cannot finalise data entry with validation errors".to_string(),
-            ErrorReference::PollingStationDataValidation,
-        ));
-    }
+    let polling_station = polling_stations_repo.get(id).await?;
+    let election = elections_repo.get(polling_station.election_id).await?;
 
-    // finalise data entry, but do not create a result yet
-    polling_station_data_entries
-        .finalise_data_entry(&mut tx, id, entry_number)
-        .await?;
-
-    if entry_number == EntryNumber::SecondEntry {
-        // get first data entry from database
-        let (_, first_data, _, _) = polling_station_data_entries
-            .get(&mut tx, id, EntryNumber::FirstEntry)
-            .await?;
-        let first_results = serde_json::from_slice::<PollingStationResults>(&first_data)?;
-
-        // only convert to result the first and second data entries are equal
-        if first_results == results {
-            polling_station_data_entries.to_result(&mut tx, id).await?
+    match entry_number {
```

Comment:
Do we want to keep the `entry_number` path parameter to this handler or let it use the entry number from the state, like in the save handler?

Reply:
If I remember correctly, we implemented it like this, because we're planning to split this endpoint into separate endpoints for first and second entry, but this doesn't break frontend functionality. Splitting the endpoints depends on how we decide to refactor the frontend, though.
The save endpoint also matches on the `entry_number` path in this PR.

Reply:
I've added it to #762.